content/automated-rust-releases.md

```diff
@@ -0,0 +1,582 @@
+        uses: MarcoIeni/release-plz-action@v0.5
+        env:
+          GITHUB_TOKEN: ${{ secrets.RELEASE_PLZ_TOKEN }}
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+```
+
+To break it down:
+
+- We are giving the workflow `write` permissions for creating pull requests.
+- We are cloning the entire git history with `fetch-depth: 0`, which is necessary to determine the next version and build the changelog.
+- We are installing Rust stable ('cause why not?!)
+- As the last step, we are running `release-plz`, heck yea!
+
+**Q**: Whoa whoa whoa hey hey hey, what's up with `GITHUB_TOKEN: ${{ secrets.RELEASE_PLZ_TOKEN }}`? Just use `${{ secrets.GITHUB_TOKEN }}` bro, which is defined as default for each repository.
+
+**A**: Good question! If you use `GITHUB_TOKEN` then it means that you are going to authenticate on behalf of GitHub Actions.
+
+**Q**: So, what's wrong with that?
+
+**A**: Well, workflows that are triggered by `on: push: tags` _won't run_ if the tag is created by GitHub Actions. So we are using a personal access token which is a way of saying "dear GitHub Actions, I am running `release-plz` and it is ME who is creating a tag so please run the further workflows". You can read more about it [here](https://release-plz.ieni.dev/docs/github/trigger).
+
+Now we are simply ready to work on our project as usual and `release-plz` will create a release PR for our changes and we can merge it to create a new release when we are ready!
+
+**Q**: What if I want to run/test `release-plz` locally though?
+
+**A**: You can if you want! The release PR is just the output of `release-plz update` command so you can run it locally and see what is changed in terms of changelog etc. Also, if you want to create a PR from the command-line you can simply run `release-plz release`!
+
+---
+
+### **cargo-dist**
+
+> [cargo-dist](https://opensource.axo.dev/cargo-dist/) makes distributing binaries easier by breaking it down to multiple steps such as plan, build, publish and announce.
+
+It is simply a tool for creating this beautiful release page with along with binary artifacts and installers:
+
+<center>
+
+![cargo-dist release](/cargo-dist-release.png)
+
+[https://github.com/orhun/daktilo/releases/tag/v0.4.0](https://github.com/orhun/daktilo/releases/tag/v0.4.0)
+
+</center>
+
+Similar to `release-plz`, `cargo-dist` is a tool that we are not going to use locally but run from the CI instead. However, it is required to be installed to set up the CI (for itself) for the first time.
+
+So let's start by installing `cargo-dist`:
+
+```sh
+cargo install cargo-dist
+```
+
+You can also use `cargo-dist` to install `cargo-dist` (i.e. the binary installer) if you wish:
+
+```sh
+cargo_dist_version="0.3.1"
+
+curl --proto '=https' --tlsv1.2 -LsSf https://github.com/axodotdev/cargo-dist/releases/download/v{version}/cargo-dist-installer.sh | sh
+```
+
+We will also be able to install our application as shown above after `cargo dist` does its magic.
```

Comment:
What do you mean with "install our application" here?

Reply:
Clarified: 920f8198954c4b6a361d3f25eb0517f5f2040ee6